src/config/strategies/multi-saml-strategy.ts

```diff
@@ -0,0 +1,32 @@
+import { MultiSamlStrategy, Profile, VerifiedCallback } from 'passport-saml';
+import setupUser from './setup-user';
+import { PassAuthAppConfig } from '../../@types';
+import { SamlOptionsCallback } from 'passport-saml/lib/passport-saml/types';
+import { Request } from 'express';
+
+export default function (config: PassAuthAppConfig): MultiSamlStrategy {
+  return new MultiSamlStrategy(
+    {
+      passReqToCallback: true,
+      getSamlOptions: (request: Request, done: SamlOptionsCallback) => {
+        const { idpId } = request.params;
+
+        const samlConfig = {
+          callbackUrl: `https://${request.headers.host}/Shibboleth.sso/SAML2/POST/${idpId}`,
+          // @ts-ignore TODO only available variable is jhu at the moment. We will have to decide wtd.
```

Comment:
Yeah I think this maybe needs some re-thinking in terms what we want to offer here anyway. I think reading in the idpId was probably geared towards a multi-tenant hosted version of pass and we need to prepare for the nearer term goal of getting a self-hosted version of the application shipped and streamlined for adoption.

Reply:
Ok, I will leave this as is for now, we can change it in the future if needed.